sdk/core/azure-core/src/main/java/com/azure/core/http/policy/UserAgentPolicy.java

```diff
@@ -73,10 +73,17 @@ public UserAgentPolicy(String sdkName, String sdkVersion, Configuration configur
         }
     }
 
+    /**
+     * Updates the User-Agent header with the value supplied in the policy.
+     *
+     * When the User-Agent header already has a value and it differs from the value used to create this policy the
+     * User-Agent header is updated by prepending the value in this policy.
+     * {@inheritDoc}
+     */
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
         String header = context.httpRequest().headers().value("User-Agent");
-        if (header == null || DEFAULT_USER_AGENT_HEADER.equals(header)) {
+        if (header == null || header.startsWith(DEFAULT_USER_AGENT_HEADER)) {
```

Comment:
Do you know why the context.httpRequest already contained this header? I'm not sure if we're covering up an underlying issue with not resetting the HTTP request context for each request

Reply:
The header information was being maintained due to the call being captured as a Mono value, every time the Mono was mapped or filtered it would re-run it again and the context was being maintained. Outside of blowing everything away on a successful completion, which just feels risky to do, I feel that making the check for adding the header stricter is safer. I had the thought of removing this if/else statement or potential making a way to indicate if the UserAgent header being added is one generated in our builder vs a customers UserAgent header.